x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractAggregationTestCase.java

```diff
@@ -435,16 +440,23 @@ private IntBlock makeGroupsVector(int groupStart, int groupEnd, int rowCount) {
      */
     private void processPageGrouping(GroupingAggregator aggregator, Page inputPage, int groupCount) {
         var groupSliceSize = 1;
+        var allValuesNull = IntStream.range(0, inputPage.getBlockCount())
+            .<Block>mapToObj(inputPage::getBlock)
+            .anyMatch(Block::areAllValuesNull);
         // Add data to chunks of groups
         for (int currentGroupOffset = 0; currentGroupOffset < groupCount;) {
-            var seenGroupIds = new SeenGroupIds.Range(0, currentGroupOffset + groupSliceSize);
+            int groupSliceRemainingSize = Math.min(groupSliceSize, groupCount - currentGroupOffset);
```

Comment:
I don't know how wasn't this failing for Vectors. But with Blocks it was trying to access values over the internal vector size

Reply:
Vectors just can't have nulls so the offsets must have lined up better.